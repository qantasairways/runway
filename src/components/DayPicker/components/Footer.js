import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'emotion';
import { Transition } from 'react-transition-group';
import {
  colours,
  layout,
  fontSize,
  fontFamily,
  fontWeight,
  mq
} from '../../../theme/airways';
import Button from '../../Button';

const wrapperStyle = {
  transform: 'translateY(100%)',
  transition: 'transform 300ms linear',
  width: '100%',
  position: 'fixed',
  bottom: 0,
  backgroundColor: colours.white,
  fontFamily: fontFamily.main
};

const wrapperTransitionStyle = {
  entering: {
    transform: 'translateY(0)'
  },
  entered: {
    transform: 'translateY(0)'
  },
  exiting: {
    transform: 'translateY(100%)'
  },
  exited: { transform: 'translateY(100%)' }
};

const innerWrapperStyle = {
  padding: `0 ${layout.gutter}`,
  width: '100%',
  maxWidth: '1030px',
  margin: '0 auto'
};

const preFooterStyle = {
  transition: 'transform 300ms linear',
  display: 'flex',
  transform: 'translateY(0)',
  flexDirection: 'column',
  justifyContent: 'center',
  padding: '11px 0',
  boxShadow: '0 -4px 6px 0 rgba(0, 0, 0, 0.08)',
  position: 'absolute',
  width: '100%',
  top: 0,
  left: 0,
  backgroundColor: colours.white,
  [mq.medium]: {
    flexDirection: 'row',
    justifyContent: 'left',
    alignItems: 'center'
  }
};

const preFooterTransitionStyle = {
  entering: {
    transform: 'translateY(-100%)'
  },
  entered: {
    transform: 'translateY(-100%)'
  },
  exiting: {
    transform: 'translateY(0)'
  },
  exited: { transform: 'translateY(0)' }
};

const preFooterTextStyle = {
  label: 'runway-footer_info-label',
  fontSize: '0.875rem',
  lineHeight: '2',
  color: colours.darkerGrey,
  [mq.medium]: {
    marginRight: '5px'
  }
};

const bottomFooterStyle = {
  padding: layout.gutter,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  position: 'relative'
};

const bottomFooterLeftCol = {
  display: 'flex',
  flexDirection: 'column',
  [mq.small]: {
    paddingRight: layout.gutter
  },
  [mq.medium]: {
    flexDirection: 'row',
    alignItems: 'center'
  }
};

const bottomFooterTextStyle = {
  label: 'runway-footer_summary-label',
  fontSize: fontSize.body,
  lineHeight: 1.56,
  [mq.medium]: {
    marginRight: '5px'
  }
};

const disclaimerStyle = {
  fontSize: '0.875rem',
  lineHeight: 1.29,
  color: colours.darkGrey
};

export function topDisclaimerStyle() {
  return css({
    label: 'runway-footer_disclaimer-label',
    ...disclaimerStyle
  });
}

// Not sure if this is needed any more?? Disclaimer will always stay on top?
export function bottonDisclaimerStyle() {
  return css({
    label: 'runway-footer_button_disclaimer-label',
    ...disclaimerStyle,
    [mq.medium]: {
      fontSize: fontSize.body
    }
  });
}

const actionButtonStyle = {
  fontWeight: fontWeight.regular,
  letterSpacing: '1.5px'
};

const Footer = ({
  summaryLabel,
  info,
  bottomFooterDisclaimer,
  preFooterDisclaimer,
  actionText,
  points,
  isCheapest,
  showPreFooter,
  showBottomFooter,
  onActionButtonClick
}) => (
  <Transition in={showBottomFooter} timeout={300}>
    {wrapperTransition => (
      <div
        css={{
          ...wrapperStyle,
          ...wrapperTransitionStyle[wrapperTransition]
        }}
      >
        <Transition in={info && showPreFooter} timeout={300}>
          {preFooterTransition => (
            <div
              css={{
                borderBottom:
                  showPreFooter && showBottomFooter
                    ? '1px solid #eeeeee'
                    : 'none',
                ...preFooterStyle,
                ...preFooterTransitionStyle[preFooterTransition]
              }}
            >
              <div css={innerWrapperStyle}>
                <span css={preFooterTextStyle}>{info}</span>
                {preFooterDisclaimer && (
                  <span css={topDisclaimerStyle()}>{preFooterDisclaimer}</span>
                )}
              </div>
            </div>
          )}
        </Transition>
        <div
          css={{
            ...innerWrapperStyle,
            ...bottomFooterStyle
          }}
        >
          <div css={bottomFooterLeftCol}>
            {summaryLabel && (
              <span css={bottomFooterTextStyle}>
                {summaryLabel}
                <span
                  css={{
                    color: isCheapest ? '#009400' : colours.darkerGrey,
                    fontWeight: fontWeight.bold
                  }}
                >
                  {points}
                </span>
              </span>
            )}
            {bottomFooterDisclaimer && (
              <span css={bottonDisclaimerStyle()}>
                {bottomFooterDisclaimer}
              </span>
            )}
          </div>
          <div>
            <Button
              label={actionText}
              cssOverrides={[actionButtonStyle]}
              onClick={onActionButtonClick}
            />
          </div>
        </div>
      </div>
    )}
  </Transition>
);

Footer.propTypes = {
  isCheapest: PropTypes.bool,
  showPreFooter: PropTypes.bool,
  showBottomFooter: PropTypes.bool,
  summaryLabel: PropTypes.string,
  points: PropTypes.string,
  info: PropTypes.string,
  preFooterDisclaimer: PropTypes.string,
  bottomFooterDisclaimer: PropTypes.string,
  actionText: PropTypes.string.isRequired,
  onActionButtonClick: PropTypes.func.isRequired
};

Footer.defaultProps = {
  isCheapest: false,
  showPreFooter: false,
  showBottomFooter: false,
  summaryLabel: '',
  points: '',
  info: '',
  preFooterDisclaimer: '',
  bottomFooterDisclaimer: ''
};

export default Footer;
