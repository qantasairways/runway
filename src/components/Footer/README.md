```js
<Footer
  preFooterInfo="Total return pricez in AUD per adult for a 25 day trip"
  bottomFooterDisclaimer=" + $344.90 ^"
  preFooterDisclaimer="^ taxes fees and carrier charges. Limited avaliability"
  actionText="Confirm"
  onActionButtonClick={() => {}}
  endDateData={{
    price: { value: 15127.0, isLowestPrice: false },
    currencySymbol: ' $'
  }}
/>
```
