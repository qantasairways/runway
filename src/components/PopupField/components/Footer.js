import React from 'react';
import PropTypes from 'prop-types';
import { generate } from 'shortid';

import { colours, fontFamily, fontWeight } from '../../../theme/airways';

import Button from '../../Button';

const textTruncationStyles = {
  whiteSpace: 'nowrap',
  overflow: 'hidden',
  textOverflow: 'ellipsis'
};

const primaryLabelContainerStyles = {
  display: 'flex',
  flexDirection: 'column',
  marginRight: '15px'
};

const primaryLabelStyles = {
  color: colours.darkerGrey,
  fontFamily: fontFamily.body,
  fontSize: '18px',
  ...textTruncationStyles
};

// eslint-disable-next-line react/prop-types
const buildLabel = ({ primaryLabels }) => {
  if (primaryLabels) {
    return (
      <div css={primaryLabelContainerStyles}>
        {primaryLabels.map((label, idx) => (
          <div
            // eslint-disable-next-line react/no-array-index-key
            key={idx}
            css={primaryLabelStyles}
          >
            {label}
          </div>
        ))}
      </div>
    );
  }
  return null;
};

const seperatorStyles = {
  borderTop: '1px solid #eeeeee'
};
const spacingStyles = { padding: '15px' };

const preFooterStyles = {
  ...spacingStyles,
  ...seperatorStyles,
  paddingTop: '9px',
  paddingBottom: '9px'
};

const buildPreFooter = preFooter =>
  preFooter && <div css={preFooterStyles}>{preFooter}</div>;

const actionButtonStyles = {
  fontFamily: 'Ciutadella',
  fontSize: '16px',
  fontWeight: fontWeight.regular,
  lineHeight: '1.5',
  letterSpacing: '1.5px',
  textAlign: 'center',
  height: '48px',
  width: '100px',
  padding: '0px 0px 0px 1px'
};

const primaryLabelOuterContainerStyles = {
  flex: 1,
  justifySelf: 'left',
  paddingRight: '15px',
  overflow: 'hidden'
};

const footerOuterContainerStyles = {
  label: 'runway-footer__root',
  zIndex: 1
};

const footerContainerStyles = {
  label: 'runway-footer__layout-container',
  ...spacingStyles,
  ...seperatorStyles,
  fontFamily: fontFamily.body,
  fontWeight: fontWeight.regular,
  background: colours.white,
  color: colours.darkerGrey,
  paddingRight: '15px',
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center'
};

const Footer = ({
  primaryLabels,
  primaryLabelAriaTitle,
  actionText,
  onAction,
  closeAriaLabel,
  preFooter
}) => {
  const summaryId = generate();

  return (
    <div css={footerOuterContainerStyles}>
      {buildPreFooter(preFooter)}
      <div css={footerContainerStyles}>
        <span
          title={primaryLabelAriaTitle}
          // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
          tabIndex={0}
          id={summaryId}
          css={primaryLabelOuterContainerStyles}
        >
          {buildLabel({ primaryLabels })}
        </span>
        <Button
          className="runway-footer__button"
          aria-label={closeAriaLabel}
          label={actionText}
          onClick={onAction}
          cssOverrides={[actionButtonStyles]}
        />
      </div>
    </div>
  );
};

Footer.propTypes = {
  primaryLabels: PropTypes.arrayOf(PropTypes.string).isRequired,
  primaryLabelAriaTitle: PropTypes.string.isRequired,
  actionText: PropTypes.string.isRequired,
  onAction: PropTypes.func,
  closeAriaLabel: PropTypes.string.isRequired,
  preFooter: PropTypes.node
};

Footer.defaultProps = {
  onAction: () => {},
  preFooter: null
};

export default Footer;
