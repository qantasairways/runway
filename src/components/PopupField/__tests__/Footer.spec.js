import React from 'react';
import { mount } from 'enzyme';
import Footer from '../components/Footer';

jest.mock('shortid', () => ({ generate: () => 'mockId' }));

describe('Footer', () => {
  const requiredProps = {
    primaryLabels: ['mock label one', 'mock label two'],
    primaryLabelAriaTitle: 'mock primaryLabelAriaTitle',
    actionText: 'mock actionText',
    closeAriaLabel: 'mock closeAriaLabel'
  };

  it('renders correctly with defaults', () => {
    const component = mount(<Footer {...requiredProps} />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with props', () => {
    const additionalProps = {
      onAction: () => {},
      preFooter: 'mock preFooter node'
    };
    const component = mount(<Footer {...requiredProps} {...additionalProps} />);

    expect(component).toMatchSnapshot();
  });
});
