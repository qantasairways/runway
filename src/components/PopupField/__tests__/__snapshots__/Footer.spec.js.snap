// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly with defaults 1`] = `
.emotion-6 {
  z-index: 1;
}

.emotion-5 {
  padding: 15px;
  border-top: 1px solid #eeeeee;
  font-weight: 400;
  background: #fff;
  color: #323232;
  padding-right: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  justify-self: left;
  padding-right: 15px;
  overflow: hidden;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-right: 15px;
}

.emotion-0 {
  color: #323232;
  font-size: 18px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-4 {
  background-color: #ee0000;
  color: #fff;
  font-family: "Ciutadella",sans-serif;
  font-size: 1rem;
  width: 100%;
  height: 3.125rem;
  border-radius: 4px;
  border-width: 0;
  padding: 14px 24px 15px;
  font-weight: 700;
  line-height: 1.56;
  -webkit-letter-spacing: 0.1em;
  -moz-letter-spacing: 0.1em;
  -ms-letter-spacing: 0.1em;
  letter-spacing: 0.1em;
  text-transform: uppercase;
  cursor: pointer;
  font-family: Ciutadella;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  text-align: center;
  height: 48px;
  width: 100px;
  padding: 0px 0px 0px 1px;
}

.emotion-4:hover {
  background-color: #ba0000;
}

.emotion-4:active {
  background-color: #ba0000;
}

.emotion-4:active {
  -webkit-transform: translate(1px,1px);
  -ms-transform: translate(1px,1px);
  transform: translate(1px,1px);
}

<Footer
  actionText="mock actionText"
  closeAriaLabel="mock closeAriaLabel"
  onAction={[Function]}
  preFooter={null}
  primaryLabelAriaTitle="mock primaryLabelAriaTitle"
  primaryLabels={
    Array [
      "mock label one",
      "mock label two",
    ]
  }
>
  <div
    className="emotion-6"
  >
    <div
      className="emotion-5"
    >
      <span
        className="emotion-3"
        id="mockId"
        tabIndex={0}
        title="mock primaryLabelAriaTitle"
      >
        <div
          className="emotion-2"
        >
          <div
            className="emotion-0"
          >
            mock label one
          </div>
          <div
            className="emotion-0"
          >
            mock label two
          </div>
        </div>
      </span>
      <Button
        aria-label="mock closeAriaLabel"
        className="runway-footer__button"
        cssOverrides={
          Array [
            Object {
              "fontFamily": "Ciutadella",
              "fontSize": "16px",
              "fontWeight": 400,
              "height": "48px",
              "letterSpacing": "1.5px",
              "lineHeight": "1.5",
              "padding": "0px 0px 0px 1px",
              "textAlign": "center",
              "width": "100px",
            },
          ]
        }
        label="mock actionText"
        onClick={[Function]}
        variant="solid"
      >
        <div
          className="runway-footer__button"
        >
          <button
            aria-label="mock closeAriaLabel"
            className="emotion-4"
            onClick={[Function]}
            type="button"
          >
            mock actionText
          </button>
        </div>
      </Button>
    </div>
  </div>
</Footer>
`;

exports[`Footer renders correctly with props 1`] = `
.emotion-7 {
  z-index: 1;
}

.emotion-6 {
  padding: 15px;
  border-top: 1px solid #eeeeee;
  font-weight: 400;
  background: #fff;
  color: #323232;
  padding-right: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  justify-self: left;
  padding-right: 15px;
  overflow: hidden;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-right: 15px;
}

.emotion-1 {
  color: #323232;
  font-size: 18px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-5 {
  background-color: #ee0000;
  color: #fff;
  font-family: "Ciutadella",sans-serif;
  font-size: 1rem;
  width: 100%;
  height: 3.125rem;
  border-radius: 4px;
  border-width: 0;
  padding: 14px 24px 15px;
  font-weight: 700;
  line-height: 1.56;
  -webkit-letter-spacing: 0.1em;
  -moz-letter-spacing: 0.1em;
  -ms-letter-spacing: 0.1em;
  letter-spacing: 0.1em;
  text-transform: uppercase;
  cursor: pointer;
  font-family: Ciutadella;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  text-align: center;
  height: 48px;
  width: 100px;
  padding: 0px 0px 0px 1px;
}

.emotion-5:hover {
  background-color: #ba0000;
}

.emotion-5:active {
  background-color: #ba0000;
}

.emotion-5:active {
  -webkit-transform: translate(1px,1px);
  -ms-transform: translate(1px,1px);
  transform: translate(1px,1px);
}

.emotion-0 {
  padding: 15px;
  border-top: 1px solid #eeeeee;
  padding-top: 9px;
  padding-bottom: 9px;
}

<Footer
  actionText="mock actionText"
  closeAriaLabel="mock closeAriaLabel"
  onAction={[Function]}
  preFooter="mock preFooter node"
  primaryLabelAriaTitle="mock primaryLabelAriaTitle"
  primaryLabels={
    Array [
      "mock label one",
      "mock label two",
    ]
  }
>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-0"
    >
      mock preFooter node
    </div>
    <div
      className="emotion-6"
    >
      <span
        className="emotion-4"
        id="mockId"
        tabIndex={0}
        title="mock primaryLabelAriaTitle"
      >
        <div
          className="emotion-3"
        >
          <div
            className="emotion-1"
          >
            mock label one
          </div>
          <div
            className="emotion-1"
          >
            mock label two
          </div>
        </div>
      </span>
      <Button
        aria-label="mock closeAriaLabel"
        className="runway-footer__button"
        cssOverrides={
          Array [
            Object {
              "fontFamily": "Ciutadella",
              "fontSize": "16px",
              "fontWeight": 400,
              "height": "48px",
              "letterSpacing": "1.5px",
              "lineHeight": "1.5",
              "padding": "0px 0px 0px 1px",
              "textAlign": "center",
              "width": "100px",
            },
          ]
        }
        label="mock actionText"
        onClick={[Function]}
        variant="solid"
      >
        <div
          className="runway-footer__button"
        >
          <button
            aria-label="mock closeAriaLabel"
            className="emotion-5"
            onClick={[Function]}
            type="button"
          >
            mock actionText
          </button>
        </div>
      </Button>
    </div>
  </div>
</Footer>
`;
